server-spi/src/main/java/org/keycloak/storage/user/UserLookupProvider.java

```diff
@@ -27,16 +27,78 @@
  * @version $Revision: 1 $
  */
 public interface UserLookupProvider {
+
```

Comment:
What do you mean? I think what is written there is still true.

Reply:
When I read it I realized I don't know how the interface `supports basic login capabilities` so I assumed it was outdated but it's probably just the fact that I don't know the application of the interface in those login capabilities.

Reply:
I see. It works because when you log in, UserStorageManager will search all storage providers, that implement UserLookupProvider, for username/email. This means it should be enough to just implement UserLookupProvider and you should be able, to login with users stored in the storage. If you want some other functionality, for example, managing users via admin console, you need to implement also UserQueryProvider. After that, when UserStorageManager searches for all users, it will consider also users from your storage because you implemented UserQueryProvider. 

In code, this is handled by methods in [AbstractStorageManager](https://github.com/keycloak/keycloak/blob/master/services/src/main/java/org/keycloak/storage/AbstractStorageManager.java), particularly in [getStorageProviderInstance(...)](https://github.com/keycloak/keycloak/blob/de09bdf32d6ede943bf2b0d60a9a452eef692129/services/src/main/java/org/keycloak/storage/AbstractStorageManager.java#L225) which will check, whether UserStorageProvider implements requested `capabilityInterface`.

Reply:
Thank for explanation.